File: TaskManager/TaskManager/UpdateIssueCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TaskManager
{
    public class UpdateIssueCommand : ICommand
    {
        private IssueRepository repository;
        private Issue newIssue, oldIssue;

        public UpdateIssueCommand(IssueRepository repo, Issue issue)
        {
            repository = repo;
            oldIssue = (Issue)repo.GetIssue(issue.Id)?.Clone();
            newIssue = issue;
        }



        public void Execute()
        {
            repository.UpdateIssue(newIssue);
        }

        public void Undo()
        {
            repository.UpdateIssue(oldIssue);
        }

        public string Description => $"Update Issue {newIssue.Id}";
    
    }
}

File: TaskManager/Lab_final_210042166/Task_Management_System/Update_Issue/UpdateIssueCommand.java
package Lab_final_210042166.Task_Management_System.Update_Issue;

import Lab_final_210042166.Task_Management_System.Interface.Command;
import Lab_final_210042166.Task_Management_System.Issue.Issue;
import Lab_final_210042166.Task_Management_System.Issue_Repository.IssueRepository;

public class UpdateIssueCommand implements Command {
    private Issue oldIssue;
    private Issue newIssue;
    private IssueRepository repository;

    @Override
    public void execute()
    {
        repository.updateIssue(newIssue);
    }
    @Override
    public void undo()
    {
        repository.updateIssue(oldIssue);
    }

}
